File: source/navigation/EnjoyFragmentNavigationDemo/app/src/main/java/com/example/xiangxue/enjoyfragmentnavigationdemo/fragment/FourFragment.java
package com.example.xiangxue.enjoyfragmentnavigationdemo.fragment;

import android.annotation.SuppressLint;
import android.os.Bundle;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.xiangxue.enjoyfragmentnavigationdemo.R;


public class FourFragment extends Fragment {


    public static Fragment newIntance() {
        FourFragment fragment = new FourFragment();
        return fragment;
    }

    public FourFragment(){
        Log.i("Zero","init: " + this.getClass().getSimpleName());
    }


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_4, container, false);
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


    }

    @Override
    public void onResume() {
        super.onResume();
        Log.i("Zero","onResume: " + this.getClass().getSimpleName());
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.i("Zero","onDestroy: " + this.getClass().getSimpleName());
    }

    public void onClick(View view) {

    }


}

File: source/jetPackDemo/navigationdemo/src/main/java/com/zero/navigationdemo/basic/FourFragment.kt
package com.zero.navigationdemo.basic

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController
import com.zero.navigationdemo.LifeCycleFragment
import com.zero.navigationdemo.R
import com.zero.navigationdemo.databinding.FragmentFourBinding
import com.zero.navigationdemo.databinding.FragmentOneBinding

class FourFragment: LifeCycleFragment() {

    private val binding by lazy{
        FragmentFourBinding.inflate(layoutInflater)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding.button.setOnClickListener {
            findNavController().navigate(R.id.action_fourFragment_to_oneFragment)
        }
        return binding.root
    }

}